src/transformers/integrations/integration_utils.py

```diff
@@ -1047,7 +1049,13 @@ def on_log(self, args, state, control, logs, model=None, **kwargs):
                         f'Trainer is attempting to log a value of "{v}" of type {type(v)} for key "{k}" as a metric. '
                         "MLflow's log_metric() only accepts float and int types so we dropped this attribute."
                     )
-            self._ml_flow.log_metrics(metrics=metrics, step=state.global_step, synchronous=False)
+
+            # "synchronous" only available with mlflow version >= 2.8.0
+            # https://github.com/mlflow/mlflow/releases/tag/v2.8.0
+            if packaging.version.parse(self._ml_flow_version) < packaging.version.parse('2.8.0'):
+                self._ml_flow.log_metrics(metrics=metrics, step=state.global_step)
+            else:
+                self._ml_flow.log_metrics(metrics=metrics, step=state.global_step, synchronous=False)
```

Comment:
Thanks for protecting this part

Reply:
This does not seem to work when `mlflow-skinny` is used. Why can't we check using `mlflow.__version__`?